### PR TITLE
Devnet deployment

### PR DESCRIPTION
**Pull request by georgysavva**
# Description of change

This PR adds automatic deployment for devnet environment.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,11 +1,32 @@
 name: Release
-'on':
-  release:
-    types:
-      - published
+on:
+  workflow_dispatch:
```
Comment (karimodm) [inline .github/workflows/release.yml:3]: Who is allowed to trigger this manually?

File: deploy/ansible/deploy.yml [hunk 2]
```diff
@@ -14,8 +14,14 @@
...
   vars:
     nodeType: goshimmer
   roles:
-    - goshimmer-node
\ No newline at end of file
+    - goshimmer-node
+
+- hosts: goshimmers:&public_nodes
+  vars:
+    nodeType: public-node
```
Comment (karimodm) [inline deploy/ansible/deploy.yml:25]: I very much like this! So the folder under `/opt` will tell you what type of node you are dealing with.

File: deploy/ansible/initial-setup.yml [hunk 3]
```diff
@@ -14,12 +14,18 @@
     - role: goshimmer-node
```
Comment (karimodm) [inline deploy/ansible/initial-setup.yml:14]: What is this file used for? Manual setup of the machines outside of the automated deployments?

File: deploy/ansible/roles/asset-registry/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    mode: '0644'
+
+- name: Stop existing services gracefully
+  community.general.docker_compose:
+    project_src: /opt/asset-registry
+    timeout: 180
+    state: present
+    stopped: yes
+
+- name: Tear down existing services
```
Comment (karimodm) [inline deploy/ansible/roles/asset-registry/tasks/main.yml:21]: This step will also try to stop the services before tearing the resources down, but I am fine with the two-steps thing!

File: deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    image: angelocapossele/asset-registry
+    command: >
+      --basicAuthUser=IOTAisCool
+      --basicAuthPassword=WeAreThereforeCool
+    ports:
+      - "80:80"
+    depends_on:
+      - mongodb
+    networks:
+      - front-tier
```
Comment (karimodm) [inline deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2:28]: The `asset-registry` also needs to be in the `back-tier` network to communicate with `mongodb`

File: deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    volumes:
+      - ./mongodb_data:/data/db
+    networks:
+      - back-tier
+
+  asset-registry:
+    image: angelocapossele/asset-registry
+    command: >
+      --basicAuthUser=IOTAisCool
+      --basicAuthPassword=WeAreThereforeCool
```
Comment (karimodm) [inline deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2:22]: These should be secrets

File: deploy/ansible/roles/drand/templates/docker-compose.yml.j2 [hunk 7]
```diff
@@ -12,9 +12,9 @@ services:
       - "0.0.0.0:{{ drandNumber }}235:8081"
       - "0.0.0.0:{{ drandNumber }}236:1236"
     environment:
-      - DRAND_PUBLIC_ADDRESS=drand-01.pre-devnet.shimmer.iota.cafe:{{ drandNumber }}234
+      - DRAND_PUBLIC_ADDRESS={{ inventory_hostname }}:{{ drandNumber }}234
     entrypoint: /usr/local/bin/entrypoint.sh
-    command: start --verbose --private-listen 0.0.0.0:8000 --public-listen 0.0.0.0:8081 --tls-disable --goshimmerAPIurl "http://drng-0{{ drandNumber }}.pre-devnet.shimmer.iota.cafe:9101" --instanceID 1 --metrics "0.0.0.0:1236"
+    command: start --verbose --private-listen 0.0.0.0:8000 --public-listen 0.0.0.0:8081 --tls-disable --goshimmerAPIurl "http://{{ goshimmerHostTemplate | replace('{drandNumber}', drandNumber) }}:9101" --instanceID 1 --metrics "0.0.0.0:1236"
```
Comment (karimodm) [inline deploy/ansible/roles/drand/templates/docker-compose.yml.j2:17]: Can we change this `goshimmerHostTemplate` to `drngHostTemplate`, as, as I understand, you only use it to iterate over the drngs hosts.

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-public-node.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,65 @@
+#jinja2: trim_blocks:True, lstrip_blocks:True
```
Comment (karimodm) [inline deploy/ansible/roles/goshimmer-node/templates/docker-compose-public-node.yml.j2:1]: I like these different compose files, even if there is a lot of duplication: it makes the content easier to read with less ifs and such.

**Review by karimodm — APPROVED** [review]
Top notch man! Well done!

File: .github/workflows/release.yml (hunk 1 above)
Comment (georgysavva) [reply .github/workflows/release.yml:3]: Everyone with write access to the repo.

File: deploy/ansible/initial-setup.yml (hunk 3 above)
Comment (georgysavva) [reply deploy/ansible/initial-setup.yml:14]: Yes. I run this playbook from my local machine to set up the whole stack. 
I decided to have it in the repo so the whole environment would be available as a code

File: deploy/ansible/roles/asset-registry/tasks/main.yml (hunk 4 above)
Comment (georgysavva) [reply deploy/ansible/roles/asset-registry/tasks/main.yml:21]: Hm, I didn't know that. 
TBH, I took that 2-steps approach from the chrysalis infrastracte repo you shared with me.

File: deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2 (hunk 5 above)
Comment (georgysavva) [reply deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2:28]: Already fixed that just forgot to push to Github

File: deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2 (hunk 6 above)
Comment (georgysavva) [reply deploy/ansible/roles/asset-registry/templates/docker-compose.yml.j2:22]: Already fixed that just forgot to push to Github

File: deploy/ansible/roles/drand/templates/docker-compose.yml.j2 (hunk 7 above)
Comment (georgysavva) [reply deploy/ansible/roles/drand/templates/docker-compose.yml.j2:17]: Yes. I like that

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-public-node.yml.j2 (hunk 8 above)
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-public-node.yml.j2:1]: Yeah I tried to balance between readability and amount of duplication.

**Review by luca-moser — APPROVED** [review]

**Pull request merged**